doc/TPC-1320-split-mobile-sdk-doc-by-platform (#249)
iadvize/documentation: docs/mobile-sdk.md

| Flutter | [Pub.dev](https://pub.dev/packages/iadvize_flutter_sdk/versions) | [GitHub](https://github.com/iadvize/iadvize-flutter-sdk) | | [Integration guide](#flutter-integration-guide) |

## Android integration guide

The iAdvize Messenger SDK for Android is available through its dedicated demo project on `Github`:

| Demo project | Latest release | API reference |
| --- | --- | --- |
| [https://github.com/iadvize/iadvize-android-sdk](https://github.com/iadvize/iadvize-android-sdk) | [https://github.com/iadvize/iadvize-android-sdk/releases/latest](https://github.com/iadvize/iadvize-android-sdk/releases/latest) | [https://iadvize.github.io/iadvize-android-sdk](https://iadvize.github.io/iadvize-android-sdk) |

### ⚙️ Setting up the SDK <span hidden>android</span>

#### 1️⃣ Adding the SDK dependency <span hidden>android</span>

Add the iAdvize repository to your project repositories inside your top-level Gradle build file:

<pre class="prettyprint">
// Project-level build.gradle.kts

allprojects {
  repositories {
    maven { url = uri("https://raw.github.com/iadvize/iadvize-android-sdk/master") }
  }
}
</pre>

> Add the iAdvize Messenger SDK dependency inside your module-level Gradle build file (replace `x.y.z` by the SDK latest version available):

Reviewer: ```suggestion
Add the iAdvize Messenger SDK dependency inside your module-level Gradle build file (replace `x.y.z` by the latest SDK version available):
```

After syncing your project you should be able to import the iAdvize dependency in your application code with `import com.iadvize.conversation.sdk.IAdvizeSDK`

⌨️ **In-context example:**

- [Project-level Gradle file](https://github.com/iadvize/iadvize-android-sdk/blob/master/build.gradle.kts)
- [Module-level Gradle file](https://github.com/iadvize/iadvize-android-sdk/blob/master/mobile/build.gradle.kts)
- [Import](https://github.com/iadvize/iadvize-android-sdk/blob/master/mobile/src/main/java/com/iadvize/conversation/sdk/demo/feature/App.kt#L5)

> *⚠️ From the version 2.5 and onward, the SDK supports video conversations using a third party native (C++) binaries. If you are delivering your app using an APK you will note a size increase as the default behavior of the build system is to include the binaries for each ABI in a single APK. We strongly recommended that you take advantage of either [App Bundles](https://developer.android.com/guide/app-bundle) or [APK Splits](https://developer.android.com/studio/build/configure-apk-splits) to reduce the size of your APKs while still maintaining maximum device compatibility.*

#### 2️⃣ Activating the SDK <span hidden>android</span>

> Before activating the SDK, you will need to provide a reference to your application object and initiate the SDK with it.

Reviewer: ```suggestion
Before activating the SDK, you will need to provide a reference to your application object and initialize the SDK with it.
```

}
</pre>

> After that you can activate the SDK using the `activate` function with your `projectId` (see the [Prerequisites](#⚙️-prerequisites) section above to get that identifier). You have access to callbacks in order to know if the SDK has been successfully activated. In case of an SDK activation failure the callback will give you the reason of the failure and you may want to retry later:

Reviewer: ```suggestion
Afterwards you can activate the SDK using the `activate` function with your `projectId` (see the [Prerequisites](#⚙️-prerequisites) section above to get that identifier). You have access to callbacks in order to know if the SDK has been successfully activated. In the case of an SDK activation failure the callback will give you the reason of the failure and you may want to retry later:
```

By default, the targeting rule language used is the user’s device current language. You can force the targeting language to a specific value using:

<pre class="prettyprint">
IAdvizeSDK.targetingController.language = SDKLanguageOption.Custom(Language.FR)
</pre>

> *⚠️ This `language` property is __NOT__ intended to change the language displayed in the SDK. It is solely used for the targeting process purpose.*

#### 2️⃣ Activating a targeting rule <span hidden>android</span>

Using a targeting rule UUID (see the [Prerequisites](#⚙️-prerequisites) section above to get that identifier), you can engage a user by calling:

<pre class="prettyprint">
IAdvizeSDK.targetingController.activateTargetingRule(
  TargetingRule(
    targetingRuleUUID,
    ConversationChannel.CHAT // or ConversationChannel.VIDEO
  )
)
</pre>

If all the following conditions are met, the default chat button should appear:

- the targeting rule exists and is enabled in the administration panel
- the targeting rule language set in the SDK matches the language configured for this rule
- an operator assigned to this rule is available to answer (connected and with a free chat slot)

> > *⚠️ After you activate a rule and it succeeds (by displaying the button), those conditions are checked every 30 seconds to verify that the button should still be displayed or not. At the first failure from this periodic check, the button is hidden and the SDK stops verifying the conditions. It means that if the rule cannot be triggered (after the first call, or after any successive check), you will have to call the `activateTargetingRule` (or `registerUserNavigation`) method again to restart the engagement process.*

Reviewer: ```suggestion
> *⚠️ After you activate a rule and it succeeds (by displaying the button), those conditions are checked every 30 seconds to verify that the button should still be displayed or not. Upon first failure from this periodic check, the button is hidden and the SDK stops verifying the conditions. It means that if the rule cannot be triggered (after the first call, or after any successive check), you will have to call the `activateTargetingRule` (or `registerUserNavigation`) method again in order to restart the engagement process.*
```

ReflectionHelpers.setStaticField(android.os.Build::class.java, "MANUFACTURER", "whatever")
</pre>

> Please also be sure to initiate the SDK during the unit tests setup (see the [Setting up the SDK](#⚙️-setting-up-the-sdk) section above).

Reviewer: ```suggestion
Please also be sure to initialize the SDK during the unit tests setup (see the [Setting up the SDK](#⚙️-setting-up-the-sdk) section above).
```
